components/abstractions/src/main/java/com/microsoft/kiota/RequestHeaders.java

```diff
@@ -0,0 +1,162 @@
+package com.microsoft.kiota;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * A class representing the headers of a request.
+ */
+public class RequestHeaders implements Map<String, Set<String>> {
+    private final HashMap<String, HashSet<String>> headers = new HashMap<>();
+    /**
+     * Adds a header to the current request.
+     * @param key the key of the header to add.
+     * @param value the value of the header to add.
+     */
+    public void add(@Nonnull final String key, @Nonnull final String value) {
+        Objects.requireNonNull(key);
+        Objects.requireNonNull(value);
+        final String normalizedKey = normalizeKey(key);
+        if(headers.containsKey(normalizedKey)) {
```

Comment:
```suggestion
        if(headers.containsKey(key)) { //containsKey implicitly normalizes the key
```
no need to normalize the keys twice

Reply:
We need the normalized key below as well